src/main/java/com/datadoghq/sketch/ddsketch/mapping/CubicallyInterpolatedMapping.java

```diff
@@ -9,6 +9,55 @@
  * A fast {@link IndexMapping} that approximates the memory-optimal one (namely {@link LogarithmicMapping}) by
  * extracting the floor value of the logarithm to the base 2 from the binary representations of floating-point values
  * and cubically interpolating the logarithm in-between.
+ * <p>
+ * Calculating the bucket index with this mapping is much faster than computing the logarithm of the value (by a factor
+ * of 3 according to some benchmarks, although it depends on various factors), and this mapping incurs a memory usage
+ * overhead of only 1% compared to the memory-optimal {@link LogarithmicMapping}, under the relative accuracy condition.
+ * In comparison, the overheads for {@link LinearlyInterpolatedMapping} and {@link QuadraticallyInterpolatedMapping} are
+ * respectively 44% and 8%.
+ * <p>
+ * Here are a few words about how to calculate the optimal polynomial coefficients.
+ * <p>
+ * The idea is that the exponent of the floating-point representation gives the floor value of the logarithm to base
+ * \(2\) of the input value for free. However, we want the logarithm to base \(\gamma = \frac{1+\alpha}{1-\alpha}\),
+ * where \(\alpha\) is the relative accuracy of the sketch. We can of course deduce that from the logarithm to the base
+ * \(2\), but that requires more than the floor value to base \(2\), and we need to actually approximate the logarithm
+ * between successive powers of \(2\). A way to do that relatively cheaply is to use the significand and to compute
+ * operations that are cheap for the CPU such as additions and multiplications. Therefore, writing \(x = 2^e(1+s)\),
+ * where \(e\) is an integer and \(s\) is between \(0\) (included) and \(1\) (excluded), we compute the index as (the
+ * floor value of) \(I_{\alpha} = m\frac{\log2}{\log\gamma}(e+P(s))\), where \(P\) is a polynomial (of degree 3 here)
+ * and \(m\) is a multiplier (\(\geq1\)) that corrects for the fact that the polynomial does not geometrically pull
```

Comment:
It's not obvious what "geometrically pull apart values from one another as well as the logarithm" means here at this point of the explanation.